### PR TITLE
test: add e2e scenarios for more gpu skus

### PR DESCRIPTION
**Pull request by lilypan26**
**What type of PR is this?**

/kind test

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: e2e/scenario/images.go [hunk 1]
```diff
@@ -10,3 +10,9 @@ var DefaultImageVersionIDs = map[string]string{
 	"marinerv2-arm64":    "/subscriptions/8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8/resourceGroups/aksvhdtestbuildrg/providers/Microsoft.Compute/galleries/PackerSigGalleryEastUS/images/CBLMarinerV2Gen2Arm64/versions/1.1699146990.16638",
 	"azurelinuxv2-arm64": "/subscriptions/8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8/resourceGroups/aksvhdtestbuildrg/providers/Microsoft.Compute/galleries/PackerSigGalleryEastUS/images/AzureLinuxV2Gen2Arm64/versions/1.1699146999.31863",
 }
+
+var DefaultGPUSeriesSKU = map[string]string{
```
Comment (cameronmeissner) [inline e2e/scenario/images.go:14]: nit: `DefaultGPUSeriesVMSizes`?

File: e2e/scenario/init.go [hunk 2]
```diff
@@ -54,4 +54,10 @@ func scenarios() []*Scenario {
 		ubuntu2204CustomCATrust(),
 		ubuntu2204ArtifactStreaming(),
 	}
+
+	for vmSeries, _ := range DefaultGPUSeriesSKU {
+		scenarios = append(scenarios, ubuntu2204gpu(vmSeries))
```
Comment (cameronmeissner) [inline e2e/scenario/init.go:59]: why not have a separate function, maybe called `gpuScenarios()`, which also returns a `[]*Scenario` - then we just concat the output of that with this slice using ellipsis operator

Comment (cameronmeissner) [reply e2e/scenario/init.go:59]: we can do that concatenation in the InitiScenarioTable func, and keep this func the same I think

Comment (cameronmeissner) [reply e2e/scenario/init.go:59]: something like `scenarios := scenarios() ... scenarios = append(scenarios, gpuScenarios()...)`

File: e2e/scenario/scenario_ubuntu2204-gpu.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import (
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+)
+
+// Returns config for the 'gpu' E2E scenario
+func ubuntu2204gpu(vmSeries string) *Scenario {
+	return &Scenario{
+		Name:        "ubuntu2204-gpu-" + vmSeries,
```
Comment (cameronmeissner) [inline e2e/scenario/scenario_ubuntu2204-gpu.go:12]: nit: fmt.Sprintf instead?

File: e2e/scenario/scenario_ubuntu2204-gpu.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+		Description: "Tests that a GPU-enabled node using an Ubuntu 2204 VHD can be properly bootstrapped",
```
Comment (cameronmeissner) [inline e2e/scenario/scenario_ubuntu2204-gpu.go:13]: should we also call out the vm size in the description?

File: e2e/scenario/images.go (hunk 1 above)
Comment (lilypan26) [reply e2e/scenario/images.go:14]: sounds good, do you think the file name should be changed? wasn't sure what a better name would be

Comment (ganeshkumar5699) [reply e2e/scenario/images.go:14]: I wonder if a separate file for this would be better - like gpuvmsizes.go or it can also be moved to e2e/const.go

Comment (cameronmeissner) [reply e2e/scenario/images.go:14]: maybe we could just do `vm.go` or similar

File: e2e/scenario/vm.go [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package scenario
+
+var DefaultGPUSeriesVMSizes = map[string]string{
```
Comment (cameronmeissner) [inline e2e/scenario/vm.go:3]: I decided to try going the `go:embed` route for the default SIG image versions map, wondering if we should also do the same here - can potential follow-up on that in a separate PR as well

File: e2e/scenario/init.go [hunk 6]
```diff
@@ -55,3 +58,13 @@ func scenarios() []*Scenario {
 		ubuntu2204ArtifactStreaming(),
 	}
 }
+
+func gpuAllVMSizeScenarios() []*Scenario {
+	scenarios := []*Scenario{}
```
Comment (cameronmeissner) [inline e2e/scenario/init.go:63]: nit: `var scenarios []*Scenario`

File: e2e/scenario/init.go [hunk 7]
```diff
@@ -7,7 +7,10 @@ import (
 // Initializes and returns the set of scenarios comprising the E2E suite in table-form.
 func InitScenarioTable(include, exclude map[string]bool) Table {
 	table := Table{}
-	for _, scenario := range scenarios() {
+	scenarios := scenarios()
+	scenarios = append(scenarios, gpuAllVMSizeScenarios()...)
```
Comment (cameronmeissner) [inline e2e/scenario/init.go:11]: nit: `scenarios := append(scenarios(), gpuAllVMSizeScenarios()...)

File: e2e/scenario/init.go [hunk 8]
```diff
@@ -55,3 +58,13 @@ func scenarios() []*Scenario {
 		ubuntu2204ArtifactStreaming(),
 	}
 }
+
+func gpuAllVMSizeScenarios() []*Scenario {
```
Comment (cameronmeissner) [inline e2e/scenario/init.go:62]: nit: `gpuVMSizeScenarios()`?

**Review by cameronmeissner — APPROVED** [review]

File: e2e/scenario/vm.go (hunk 5 above)
Comment (lilypan26) [reply e2e/scenario/vm.go:3]: deferring for now

**Pull request merged**